tests/tests.py

```diff
@@ -149,97 +149,160 @@ def test_rds(self):
+                        {'ToPort': 443,
+                         'IpProtocol': 'tcp',
+                         'CidrIp': '0.0.0.0/0',
+                         'FromPort': 443},
+                        {'ToPort': 80,
+                         'IpProtocol': 'tcp',
+                         'CidrIp': '0.0.0.0/0',
+                         'FromPort': 80}
+                    ],
+                    'VpcId': {'Ref': 'VPC'},
+                    'GroupDescription': 'DefaultELBSecurityGroup'
+                },
+                'Type': u'AWS::EC2::SecurityGroup',
+            },
+            'DefaultSGtestdevinternal': {
+                'Properties': {
+                    u'SecurityGroupIngress': [
+                        {'ToPort': 443,
+                         'IpProtocol': 'tcp',
+                         'CidrIp': '0.0.0.0/0',
+                         'FromPort': 443},
+                        {'ToPort': 80,
+                         'IpProtocol': 'tcp',
+                         'CidrIp': '0.0.0.0/0',
+                         'FromPort': 80}
+                    ],
+                    'VpcId': {'Ref': 'VPC'},
+                    'GroupDescription': 'DefaultELBSecurityGroup'
+                },
+                'Type': 'AWS::EC2::SecurityGroup',
+            }
+        }
 
         config = ConfigParser(
             ProjectConfig(
                 'tests/sample-project.yaml',
                 'dev').config,'my-stack-name')
         elb_cfg, elb_sgs = config.elb()
-        compare(known, elb_cfg)
 
-    def test_elb_no_sgs(self):
-        from testfixtures import compare
+        compare(expected_resources, elb_cfg)
+
+        compare(expected_sgs, elb_sgs)
+
+    def test_elb_custom_sg(self):
+
+        expected_sgs = {
+            'SGName': {
+                'Properties': {
+                    u'SecurityGroupIngress': [
+                        {'ToPort': 443,
+                         'IpProtocol': 'tcp',
+                         'CidrIp': '1.2.3.4/32',
+                         'FromPort': 443},
+                    ],
+                    'VpcId': {'Ref': 'VPC'},
+                    'GroupDescription': 'DefaultELBSecurityGroup'
+                },
+                'Type': u'AWS::EC2::SecurityGroup',
+            },
+        }
+
+        project_config = ProjectConfig('tests/sample-project.yaml', 'dev')
+
+        # Remove the "test-dev-internal" ELB
+        project_config.config['elb'].pop()
+        project_config.config['elb'][0]['security_groups'] = {
+            'SGName': [
+                {'IpProtocol': 'tcp',
+                 'FromPort': 443,
+                 'ToPort': 443,
+                 'CidrIp': '1.2.3.4/32'},
+            ]
+        }
+        config = ConfigParser(project_config.config, 'my-stack-name')
+        elb_cfg, elb_sgs = config.elb()
+
+        compare(expected_sgs, elb_sgs)
+
+        [elb] = (e.values()[0] for e in elb_cfg if e.values()[0]['Type'] == 'AWS::ElasticLoadBalancing::LoadBalancer')
```

Comment:
My thought for doing this is that: "I'm not testing the whole return value (which is about a 40 line block) but just this one specific property (And to not overly test/retest that which is already covered in `test_elb`)

Is this clearer:

```python
        [elb] = (e.values()[0] for e in elb_cfg if e.has_key('ELBtestdevexternal'))

```